### PR TITLE
Introducing IR query service

### PR DESCRIPTION
**Pull request by BingqingLyu**
## Related issue number

The IR layer for GAIA, integrated with groot, vineyard and the whole GraphScope pipeline, which brings in:
* Support match()-step in Gremlin, for graph pattern matching
* A completely redesigned IR layer to decouple the query language dependency, and further for query optimization
* Define the supported Gremlin's grammar via Antlr
* More Gremlin's Grammer supports, including two extended Grammer, expr() for composing expression more conveniently, out/in/both(x..y).endV() for path expansions.

## Related issue number

#1377

### CODE REVIEW

**Review by longbinlai — COMMENTED** [review]
Please add IR related CIs

File: research/engine/pegasus/server-v0/examples/dyn_lib_server.rs.bak [hunk 1]
```diff
@@ -0,0 +1,281 @@
+#[macro_use]
```
[the remark this answers is not in the document]
Comment (BingqingLyu) [reply research/engine/pegasus/server-v0/examples/dyn_lib_server.rs.bak]: Done

**Comment by shirly121** [comment]
> Please add IR related CIs

Tests including unit tests (mvn test & cargo test), gremlin ci tests on experimental store & groot store are all added. See https://github.com/alibaba/GraphScope/pull/1407/files#diff-01c8a5027542b364af96fa8b4c5871e3b4f893cd87d2ebdbb86adfd5b2e72699 for updates.

**Review by longbinlai — APPROVED** [review]
LGTM. Code factor issue is due to the old python codes. Can ignore? @sighingnow

**Comment by longbinlai** [comment]
@yecol If no other issue, can approve this? Codefactor issue is due to some legacy python codes according to @BingqingLyu , which can be ignored?

**Comment by yecol** [comment]
Sure!
LGMT~

**Review by yecol — APPROVED** [review]

**Pull request merged**